framework/mongo/utils.py

```diff
@@ -63,10 +63,18 @@ def wrapper(cls):
 
 def get_or_http_error(Model, pk):
     instance = Model.load(pk)
+
     if getattr(instance, 'is_deleted', False):
-        raise HTTPError(http.GONE, data=dict(
-            message_long="This resource has been deleted"
-        ))
+        if hasattr(instance, 'registrations'):
```

Comment:
But why is it necessary? I don't believe the issue has to do with registrations at all--whether you link to a deleted project from a registration or just go to the deleted project from your browser, you should still see the same error page.